code/controllers/Processes/explosives.dm

```diff
@@ -0,0 +1,200 @@
+var/datum/controller/process/explosives/bomb_processor
+
+// yes, let's move the laggiest part of the game to a process
+// nothing could go wrong -- Lohikar
+/datum/controller/process/explosives
+	var/list/work_queue
+	var/ticks_without_work = 0
+	var/powernet_rebuild_was_deferred_already
+
+/datum/controller/process/explosives/setup()
+	name = "explosives"
+	schedule_interval = 5 // every half-second
+	work_queue = list()
+	bomb_processor = src
+
+/datum/controller/process/explosives/doWork()
+	if (!work_queue)
+		setup()	// fix for process failing to re-init after termination
+
+	if (!(work_queue.len))
+		ticks_without_work++
+		if (ticks_without_work > 5)
+			if(defer_powernet_rebuild)
+				makepowernets()
+				defer_powernet_rebuild = 0
+		return
+
+	ticks_without_work = 0
+
+	for (var/datum/explosiondata/data in work_queue)
```

Comment:
Minor optimization point.

Since you're expecting work_queue to only contain datum/explosion data, you can forego a istype() check and typecast implicitly:
```
for (var/A in work_queue)
    var/datum/explosiondata/data = A
```